### ISSUE TITLE
Playing a move on an empty board throws an error

### ISSUE THREAD

**Issue by LandonSchropp** [jhlywa/chess.js#282]
If you try to play a move from an empty position, chess.js throws an error.

``` js
let chess = new Chess("8/8/8/8/8/8/8/8 w KQkq - 0 1");
chess.move("e4");
```

This produces the following error:

```
chess.js:520 Uncaught TypeError: Cannot read property 'type' of undefined
    at add_move (chess.js:520)
    at generate_moves (chess.js:635)
    at move_from_san (chess.js:1140)
    at Object.move (chess.js:1771)
```

I would instead expect this to return `null`.

**Comment by jhlywa** [comment]
Thanks for the report.  This was fixed in `0.12.0`

**Comment by LandonSchropp** [comment]
🎉 👍  Thanks!
